### DISCUSSION TITLE
Determine if WebP is lossy or lossless

### DISCUSSION THREAD

**Discussion by avalonv** [python-pillow/Pillow#6716] [Q&A] [answered]
I'm developing an application for optimizing comics and knowing the original format is somewhat important, since converting from lossless to lossy formats can often increase file size. Currently I try to convert both and then compare their sizes, but is there a built-in way to check?

**Comment by radarhere** [comment, marked as answer]
A built-in way? No.

However, I saw https://github.com/python-pillow/Pillow/blob/aedbc77618b83f8486174f6b3f646651d3f04a55/src/PIL/WebPImagePlugin.py#L17-L29

and think that the following should be able to tell you if the WebP file is lossless or not.

```python
with open("Tests/images/hopper.webp", "rb") as fp:
    lossless = fp.read(15)[-1:] == b"L"
```

**Reply by avalonv** [reply]
I apologize for how long it took me to get to it, but this isn't really working for me :(

Do images created by Pillow not have that byte set?

**Reply by avalonv** [reply]
Would it help if I added a sample file for you to test on your machine?

**Reply by radarhere** [reply]
This is also mentioned in the [specification](https://developers.google.com/speed/webp/docs/riff_container#simple_file_format_lossless), so I'm surprised that it isn't working for you.

If you could attach sample files, that would be great. Also if you could mention how you're figuring out whether they are lossy or lossless.

**Reply by avalonv** [reply]
Well, I'm *fairly* sure they're lossless because I'm doing batch image processing which checks the size of files at the end, and ones generated with the Image.save() lossless parameter set to true are clearly larger (and they don't really change size on repeated saves).

Here's a file https://www.dropbox.com/s/jx5ti5oavookqil/berk2.webp

Here's the relevant line I used to create it: 

https://github.com/avalonv/reCBZ/blob/bee82c61c5864f8c753567c9722a3715fc93a696/src/reCBZ/formats.py#L45

**Reply by radarhere** [reply]
Oh. I think I just made a simple error in my first comment, writing 15 instead of 16.

Try
```python
with open("berk2.webp", "rb") as fp:
    lossless = fp.read(16)[-1:] == b"L"
```

**Reply by avalonv** [reply]
```py
c>>> with open('berk2.webp', 'rb') as fp:
c...    lossless = fp.read(16)[-1:] == b"L"
c... 
c>>> print(lossless)
True
```

You're a lifesaver, thank you!
